### DISCUSSION TITLE
What is the intended way to "enqueue" operations from IDynamicPeripheral#callMethod

### DISCUSSION THREAD

**Discussion by pupnewfster** [cc-tweaked/CC-Tweaked#728] [Q&A] [answered]
The javadocs for `IDynamicPeripheral#callMethod` specifies that it is called from the ComputerCraft Lua thread and that the passed in ILuaContext can be used to wait for events or otherwise yield. The issue I am running into is there does not seem to be a clean manner to do this only using the API context without just creating a copy of ComputerCraft's `TaskCallback` class which is in `dan200.computercraft.core.asm`. Is it intended that modders basically implement a copy of this class given it is not in the API so in theory could change in a breaking manner at any time, or is this a class that modders should feel safe in using?

Either way from an API standpoint what may make the most sense would be to add a helper method in `ILuaContext` that basically wraps/returns the result of `TaskCallback.make(ILuaContext, ILuaTask)` so that modders can safely and easily enqueue tasks to be run on the server thread purely from the API without leaking any implementation details into the API.

**Comment by SquidDev** [comment, marked as answer]
Yep, copying the `TaskContext` class is probably the easiest solution for now. Yep, I guess it might be sensible to make it a method on `ILuaContext` - I think my assumption was that if people were using `IDynamicPeripheral`, people were probably rolling their own everything else anyway...
